packages/eslint-plugin/src/rules/shorthand-property-sorting/__tests__/rule.test.ts

```diff
@@ -0,0 +1,240 @@
+        borderColor: '#00b8d9',
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+      errors: [{ messageId: 'shorthand-first' }],
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `incorrect property ordering -> nested ObjectExpression (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const containerAppearance = {
+        default: ${call}({
+          borderBlockEnd: '1px solid #00b8d9',
+          borderBlock: '#00b8d9',
+          border: '#00b8d9',
+        }),
+        success: ${call}({
+          border: '#00b8d9',
+          borderBlock: '#00b8d9',
+          borderBlockEnd: '1px solid #00b8d9',
+        }),
+        inverse: ${call}({
+          border: '#00b8d9',
+          borderBlockEnd: '1px solid #00b8d9',
+          borderBlock: '#00b8d9',
+        }),
+      };
+    `,
+      errors: [{ messageId: 'shorthand-first' }, { messageId: 'shorthand-first' }],
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `incorrect property ordering -> 6 reordering errors (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        borderTop: '1px solid #00b8d9',
+        border: '#00b8d9',
+        borderColor: '#00b8d9',
+        borderRight: '#00b8d9',
+        gridTemplate: '1fr 1fr',
+        gridRow: '1',
+        borderBlockStart: '10px',
+      });
+      export const EmphasisText = ({ children }) => <span css={styles}>{children}</span>;
+    `,
+      errors: [{ messageId: 'shorthand-first' }],
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `includes pseudo-selectors -> pseudo is out of order (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        '&:hover': {
+            borderTop: '1px solid #00b8d9', // 4
+            borderColor: 'red', // 2
+            border: '1px solid #00b8d9', // 1
+        },
+        border: '1px solid #00b8d9', // 1
+        borderColor: 'red', // 2
+        borderTop: '1px solid #00b8d9', // 4
+      })
+      `,
+      errors: [{ messageId: 'shorthand-first' }],
+    })),
+    ...packages_calls_and_imports.map(([pkg, call, imp]) => ({
+      name: `includes pseudo-selectors -> non-pseduo are out of order (${pkg}: '${imp}')`,
+      code: outdent`
+      import {${pkg}} from '${imp}';
+      const styles = ${call}({
+        '&:hover': {
+          border: '1px solid #00b8d9', // 1
+          borderColor: 'red', // 2
+          borderTop: '1px solid #00b8d9', // 4
+        },
+        borderTop: '1px solid #00b8d9', // 4
+        borderColor: 'red', // 2
+        border: '1px solid #00b8d9', // 1
+      })
+      `,
+      errors: [{ messageId: 'shorthand-first' }],
+    })),
+
+    /* fixer can't deal with nested fixing in one go. I've split this test into:
```

Comment:
Is this comment still required?

Reply:
Yes, I that test can't deal with the whole case the first time round and I had to split it in 2, it looks confusing without that context

Reply:
I thought we remove the autofixer